src/OpenLoco/src/Paint/PaintBridge.cpp

```diff
@@ -0,0 +1,1410 @@
+
+            const int16_t lhsSupportLenght = unkHeight - k4F919C[slope];
+            if (unkHeight < 0)
+            {
+                return std::nullopt;
+            }
+            return SupportLengths{ lhsSupportLenght, rhsSupportLength };
+        }();
+        if (supportLengths.has_value())
+        {
+            // 0x0042BD9C
+
+            const auto& [lhsSupportLenght, rhsSupportLength] = supportLengths.value();
+            const auto supportHeight = bridgeEntry.height - bridgeObj.deckDepth;
+
+            auto image = bridgeEntry.imageBase.withIndex(bridgeObj.image).withIndexOffset(Bridge::ImageIds::deckBaseWithSupportHeaderEdge3);
+            auto offset = baseHeightOffset - World::Pos3{ 0, 0, bridgeObj.deckDepth };
+            World::Pos3 bbOffset = { 0, 0, 14 };
+            World::Pos3 bbLength = { 32, 32, 1 };
+            session.addToPlotList4FD150(image, offset, bbOffset + offset, bbLength);
+
+            paintSupportDiagonal(session, bridgeObj, bridgeEntry, lhsSupportLenght, rhsSupportLength, supportHeight);
+        }
+        else
+        {
+            // 0x0042BE22
+            auto image = bridgeEntry.imageBase.withIndex(bridgeObj.image).withIndexOffset(Bridge::ImageIds::deckBaseNoSupportEdge3);
+            auto offset = baseHeightOffset - World::Pos3{ 0, 0, 16 };
+            World::Pos3 bbOffset = { 0, 0, 14 };
+            World::Pos3 bbLength = { 32, 32, 1 };
+            session.addToPlotList4FD150(image, offset, bbOffset + offset, bbLength);
+        }
+        // 0x0042BE77
+
+        if (bridgeEntry.subType == 0)
+        {
+            auto wallImage = bridgeEntry.imageBase.withIndex(bridgeObj.image).withIndexOffset(Bridge::ImageIds::deckWallEdge3);
+            World::Pos3 bbOffset = { 17, 17, 2 };
+            World::Pos3 bbLength3 = { 2, 2, 24 };
+            session.addToPlotList4FD150(wallImage, baseHeightOffset, bbOffset + baseHeightOffset, bbLength3);
+        }
+    }
+
+    // 0x0042B4DC
+    static void paintBridgeNE(PaintSession& session, const BridgeObject& bridgeObj, const BridgeEntry& bridgeEntry, const int16_t supportLength, const uint8_t slope, const uint8_t pillarSpacing, const SpanImages& spanImages)
+    {
+        const auto baseHeightOffset = World::Pos3{ 0, 0, bridgeEntry.height };
+        if ((bridgeObj.flags & BridgeObjectFlags::hasRoof) != BridgeObjectFlags::none)
+        {
+            const auto roofImageIdx = (bridgeEntry.edgesQuarters & 0xF0) == 0xF0 ? Bridge::ImageIds::roofFullTile : spanImages.roof;
+            auto roofImage = bridgeEntry.imageBase.withIndex(bridgeObj.image).withIndexOffset(roofImageIdx);
+            World::Pos3 bbOffset = { 0, 0, 30 };
+            World::Pos3 bbLength = { 32, 32, 0 };
+            session.addToPlotList4FD150(roofImage, baseHeightOffset, bbOffset + baseHeightOffset, bbLength);
+        }
+
+        auto supportLengths = [&session, &bridgeEntry, &bridgeObj, pillarSpacing, supportLength, slope]() -> std::optional<SupportLengths> {
```

Comment:
I think we should look into de-duplicating this one. The difference I see is the mask and the indices so we can probably just make those additional (template) parameters, The runtime overhead will be probably negligible if done as normal arguments.

Reply:
There are definitly a good few areas that could be deduplicated but i think it best to address in a follow on. Perhaps after we have some tests for this.